hardware_integration/src/can_hardware_interface.cpp

```diff
@@ -376,7 +376,8 @@ void CANHardwareInterface::can_thread_function()
 		threadConditionVariable.wait(lMutex);
 		CanHardware *pCANHardware;
 
-		if (threadsStarted)
+		if ((threadsStarted) &&
+		    (nullptr != pCANHardware))
```

Comment:
Isn't pCANHardware always a nullptr when it reaches this if statement?

Reply:
Might want to add the `nullptr != pCANHardware` check to line 384 instead?

Reply:
Ah yep, haha, I guess I need more coffee yet this morning

Reply:
Though I feel like `hardwareChannels` shouldn't hold on to `nullptr`s in general in the future?

Reply:
Some amount of nullptr's might be inevitable, like if someone creates 3 can channels, then destroys the one at index 0, we would want the indices to remain the same, so leaving the dead channel as nullptr makes some sense I think. In this case, I'm not entirely sure where the nullptr is coming from yet, as there is an object assigned before start is called.

Reply:
I'm also not seeing the crash on Windows with PCAN, only on Linux with socket CAN... more investigation needed

Reply:
> Some amount of nullptr's might be inevitable, like if someone creates 3 can channels, then destroys the one at index 0, we would want the indices to remain the same, so leaving the dead channel as nullptr makes some sense I think. In this case, I'm not entirely sure where the nullptr is coming from yet, as there is an object assigned before start is called.

Hmm, I see. The indices are only needed for CANHardwareFrame's channel index assignments, which itself is needed for logic within CANNetwork, right? 

I suggest that in the future we can remove this limitation by assigning a weak_ptr of a CANNetwork object to the CANHardware struct, which contains the info for each hardware channel. (Though this can only be done when and if we deviate from the singleton pattern for CANNetwork I think). Would that work?

Reply:
Yeah, if we want to be free of that limitation then as you said we'll need to map between CANNetwork channels and hardware channels, and I think that would work OK?

Reply:
We'll see I guess. It seems like a lot of work for a small improvement.